extensions/libraries/redcore/controller/admin/base.php

```diff
@@ -296,11 +296,9 @@ protected function getRedirectToListRoute($append = null)
 	{
 		$returnUrl = $this->input->get('return', '', 'Base64');
 
-		if ($returnUrl)
+		if ($returnUrl && JURI::isInternal(base64_decode($returnUrl)))
 		{
-			$returnUrl = base64_decode($returnUrl);
-
-			return JRoute::_($returnUrl . $append, false);
+			return JRoute::_(base64_decode($returnUrl), false);
```

Comment:
so you are removing $append variable at the end... have you test this on Vanir maybe?
imo it would be best if you load everything in the JUri object, and add $append there, so you wont have any duplicates in the URI fields, and you would not break this feature if some extension is using it

Reply:
it's bad design imo to modify the return url that way, but i won't push for this fix as long as it doesn't bother others.